docs/setup/upgrade/upgrade-migrations.asciidoc

```diff
@@ -14,19 +16,21 @@ WARNING: The following instructions assumes {kib} is using the default index nam
-
-While {kib} is starting up and before serving any HTTP traffic, it checks to see if any internal mapping changes or data transformations for existing saved objects are required.
+The index aliases `.kibana` and `.kibana_task_manager` will always point to the most up-to-date version indices.
 
-When changes are necessary, a new migration is started. To ensure that only one {kib} instance performs the migration, each instance will attempt to obtain a migration lock by creating a new `.kibana_N+1` index. The instance that succeeds in creating the index will then read batches of documents from the existing index, migrate them, and write them to the new index. Once the objects are migrated, the lock is released by pointing the `.kibana` index alias the new upgraded `.kibana_N+1` index. 
+The first time a newer {kib} starts, it will first perform an upgrade migration before starting plugins or serving HTTP traffic. To prevent losing acknowledged writes old nodes should be shutdown before starting the upgrade. To reduce the likelyhood of old nodes losing acknowledged writes, {kib} 7.12+ will add a write block to the outdated index.
 
-Instances that failed to acquire a lock will log `Another Kibana instance appears to be migrating the index. Waiting for that migration to complete`. The instance will then wait until `.kibana` points to an upgraded index before starting up and serving HTTP traffic.
+When several {kib} instances are connected to the same {es} cluster and upgraded, all {kib} instances will participate in the upgrade migration in parallel.
 
-NOTE: Prior to 6.5.0, saved objects were stored directly in an index named `.kibana`. After upgrading to version 6.5+, {kib} will migrate this index into `.kibana_N` and set `.kibana` up as an index alias. +
-Prior to 7.4.0, task manager tasks were stored directly in an index name `.kibana_task_manager`. After upgrading to version 7.4+, {kib} will migrate this index into `.kibana_task_manager_N` and set `.kibana_task_manager` up as an index alias.
+[NOTE]
+====
+ - Prior to 6.5.0, saved objects were stored directly in an index named `.kibana`. After upgrading to version 7.12+, {kib} will migrate this index into `.kibana_{kibana_version}_001` and set `.kibana` up as alias to this index. +
+ - Prior to 7.4.0, task manager tasks were stored directly in an index name `.kibana_task_manager`. After upgrading to version 7.12+, {kib} will migrate this index into `.kibana_task_manager_{kibana_version}_001` and set `.kibana_task_manager` up as an index alias. +
+ - Prior to 7.12.0, saved objects and task manager tasks were stored in `.kibana_N` and `.kibana_task_manager_N` where N is a number that's increased every time an upgrade migration was performed. After upgrading to version 7.12+, {kib} will migrate these indices into `.kibana_{kibana_version}_001` and `.kibana_task_manager_{kibana_version}_001` respectively.
```

Comment:
Instead of the "Prior to X" language, I think it would be more clear if we said which versions each statement applied to, example:

- In Kibana versions 4.0.0 through 6.4.2, ...
- In Kibana versions 6.5.0 through 7.3.2, ...
- In Kibana versions 7.4.0 through 7.11.2, ...

Reply:
I tried to condense this into a table

Reply:
<img width="750" alt="Screenshot 2021-01-22 at 15 27 24" src="https://user-images.githubusercontent.com/1281646/105503046-66a06600-5cc6-11eb-9b71-8283be43d886.png">